arch/x86/entry.S

```diff
@@ -208,6 +208,26 @@ ENTRY(syscall_handler_entry)
     SYSRET
 END_FUNC(syscall_handler_entry)
 
+ENTRY(sysenter_handler_entry)
+    push %_ASM_CX
+    push %_ASM_DX
+    syscall_from_usermode
+
+    cmp $SYSCALL_EXIT, %_ASM_AX
+    jz syscall_exit
+
+    mov %_ASM_DI, %_ASM_DX
+    mov %r10, %_ASM_CX
+    mov %_ASM_AX, %_ASM_DI
+    call syscall_handler
+
+    syscall_to_usermode
+    pop %_ASM_DX
+    pop %_ASM_CX
```

Comment:
Should we sanity check DX and CX somehow, so uerland can't make them point to arbitrary addresses, e.g. into the non-cannonical address space?

Reply:
If we want to do this, the correct range for usermode would be: `VIRT_USER_BASE` - `0x00007FFFFFFFFFFF`

Reply:
I think it's not too bad to just let it crash if userland provides bad values. I don't know if for example masking them to valid userland addresses would make such a crash less cryptic..

Reply:
So then, it would be to compare and raise and error already here... Since will no longer expose the low-level sysenter(), maybe there's no need to care?

Reply:
Well, it matters for *how* it crashes. When RCX or RDX are non-canonical, SYSEXIT will #GP while still in kernel mode instead of, e.g., #PF back in userland when RIP is canonical but not mapped. Slightly different semantics, maybe worth to be aware of?
Not to mention SYSEXIT will #UD on AMD when executed in long mode.

Reply:
oh yes then it makes absolute sense. I assumed the crash would always happen in user mode.
The easy way out is to just mask it.. Is that enough or better terminate the usermode task somehow?

sysexit: yes, and sysenter too.. Do we want to warn AMD-users about this?

Reply:
I agree, let's handle it separately.